### ISSUE TITLE
Usability: default theme is too low contract and too blurry

### ISSUE THREAD

**Issue by JonasT** [atom/atom#13527]
When I installed and opened up Atom for the first time, I was put off a bit: the default theme is too low contract and badly readable. It seems to be made mainly to look super fancy, while throwing basic readability out of the window: especially on the tabs, the text is too small, too blurry and too low contrast.

Compare this to Visual Studio Code which has a similarly nice color scheme, but unlike Atom's default colors Visual Studio Code has super crisp and high contrast text on the tabs:

![contrast_comparison](https://cloud.githubusercontent.com/assets/1573352/21582240/042f260c-d050-11e6-8324-eb2f494d1b30.png)

As you can see, even the file menu is just a tad more crisp and more comfortably readable with Visual Studio Code.

Because this might give an initial bad impression (it certainly did for me) and potentially needlessly uses up people's time for altering the color scheme and fonts to something more readable, I suggest that this should be changed for a future version of Atom.

**Comment by JonasT** [comment]
Just for what it's worth:

Gitlab (and possibly other software) allows easy maintenance of a separate backlog list made of the issues on the tracker, which means tickets can be easily labeled in a way that keeps them out of the backlog. I don't know if Github already offers something similar, but if it does you might want to look into that!

(not meant as a criticism, just a suggestion since I have some professional experience with agile development and related software)

**Comment by arcticicestudio** [comment]
Are you sure that this is not related to your setup?
The One Dark UI theme is designed to be responsive. It automatically resizes the UI [based on the window size](https://github.com/atom/one-dark-ui/blob/master/styles/config.less#L15#L23).

I tried to reproduce your screenshot using adjusted settings for a larger UI in a Atom window that is exactly half of the size of my screen, but with these settings it works just fine relating to the tab- and font size.

![ghi-13527-scrot-window](https://cloud.githubusercontent.com/assets/7836623/21586058/3aaa5f72-d0cc-11e6-83c9-c8e7a8a06e41.png)

![ghi-13527-scrot-settings](https://cloud.githubusercontent.com/assets/7836623/21586060/3d66884e-d0cc-11e6-92c9-6cb79df28924.png)

About the blurry font: It looks like you're running Linux and I'm running Arch Linux on a GNOME DE as well. The blurry font may be related to your system font rendering lib (your window titles are looking deformed too), you should try to [configure your fonts](https://wiki.archlinux.org/index.php/Font_configuration) or install a improved patched lib like [Infinality](https://wiki.archlinux.org/index.php/Infinality).

**Comment by JonasT** [comment]
The tabs in your screenshot looks similarly low contrast to me, it just seems to have a slightly bigger font size (which I guess might be a DPI issue?).

As for the comparatively blurry font, I'd just like to note that given the same base system, Visual Studio Code clearly has a crisper more readable (IMHO font on the tabs - including also compared to all of your screenshots. Therefore, it seems to me like it should be possible to ship with a font and/or render lib in a way that gets those better results.

Microsoft Windows is known for having fonts that are more geared towards pixel-exact readable results (unlike the often more blurry but supposedly prettier fonts on e.g. Mac OS X), so maybe that is related to their choice. Whatever the rationale and technical stuff for the font rendering is, Atom definitely looks too blurry and low contrast for me to be easily readable (including in all of your screenshots).

TL;DR: I don't know if you want to argue that your screenshots look better, but apart from maybe font size they look similarly bad (in my personal opinion about the color choices)

**Comment by 50Wliu** [comment]
VS Code might be using a newer Electron version.  Generally, the blurriness/crispness of the font is dependent on Electron version.  For example, everything is grayscale-antialiased for me on Windows using Electron 1.3, but is subpixel-antialiased using 1.4 (which we're working to get Atom on).

Regarding the backlog comments: I don't think GitHub has something like that.  We can label issues so that they don't get lost in the closed list however, which is what I did for this one.  There's also the newly-introduced Projects, but we're still experimenting with the best way to use it.

**Comment by simurai** [comment]
I can reproduce the "blurriness" of the tabs also on macOS. I think it's because it uses a mask to fade out the text (when it gets too long) and that disables the sub-pixel antialiasing. I'll have to look into it to be sure.

About the contrast. This is quite hard to get right because people have different preferences. We once tried to increase contrast for the syntax theme and then people wanted the lower contrast back. There is the argument that in general or "out of the box" it's better to have too high contrast than too low. Because with too high contrast you can still read, but with too low not. Something we should keep in mind for the next major version.

**Comment by JonasT** [comment]
I definitely do not mind low contrast alternatives shipped out of the box. I also agree that the current theme looks cool and sleek! It's just that I think on half of my screens it's not very readable... (especially compared to Visual Studio Code, which still looks somewhat sleek and modern)

The old original Atom theme is better, but of course it looks a bit dated in comparison, especially the tabs.

**Comment by JonasT** [comment]
As a more detailed feedback, I have now found a config which I don't find absolutely unbearable to use and which still looks pretty:

UI theme: "One Dark"
Syntax theme: "Atom Dark"
Additional custom styles.less changes:
```
atom-text-editor::shadow .comment {
  color:#eeaa75;
}

.tab-bar .tab .title {
   color:#aabbbb;
}

.tab-bar .tab.active .title {
   color:#aaffee;
}
```

Explanation for styles.less changes:

The UI theme "One Dark" has way too low contrast on the tabs, while the old "Atom Dark" theme has way too low contrast in the editor settings and the tabs look a bit ugly. (hence "One Dark" UI theme + custom .tab-bar changes to increase contrast on tabs) The syntax theme "One Dark" is overall way too low contrast, while "Atom Dark" is ok but the comments are too hard to read. (hence "Atom Dark" syntax theme + custom .comment changes)

Maybe someone wants to take a look at that configuration and figure out if the next Atom theme can be a better mix of stylish and usable/readable. Not everyone has the fanciest super high brightness screens where low contrast themes work nicely... (and I do have good bright screens on some of my PCs where the default theme isn't still particularly well-readable but it's ok, but on my older desktop monitor it's an absolute disaster)

**Comment by 50Wliu** [comment]
If you want some more contrast and also like One Dark, try One Dark Vivid.

**Comment by JonasT** [comment]
@50Wliu I could only find a syntax theme "One Dark Vivid" which doesn't change much. The problematic points of the tab font being way too dark or the comments being too greyish unobtrusive unreadable aren't really addressed by it.

**Comment by simurai** [comment]
@JonasT Seems your text gets rendered quite thin. Here a screenshot on macOS which is thicker:

![screen shot 2017-01-17 at 5 03 07 pm](https://cloud.githubusercontent.com/assets/378023/22012573/10c86fa6-dcd8-11e6-89cc-58cc922163e9.png)

Also, notice that I have "Enable VCS coloring" enabled (in the tabs settings). The low contrast of unchanged files helps the rest to stand out more. But not everyone has that enabled. I agree to keeping accessibility in mind for One Next.

Contrast for comments is also interesting. Some people like low contrast for comments to make code "temporarily disappear". Like disabling some blocks/lines of code and you don't really need to read because it gets skipped. On the other hand some people like higher contrast because comments also get used for documentation that should be easy to read.

**Comment by JonasT** [comment]
@simurai comments are definitely NOT meant mainly for making code disappear. If that is what you mainly do with them instead of actually *commenting your code for understanding*, you are using them wrong.

As for text rendering, I am not disputing that maybe text rendering problems are making the problems I am seeing worse. However, Visual Studio Code definitely also uses way higher contrast for most colors and I like that a lot more, so I think it's at least a mix of the two issues (text rendering as well as too low contrast choices for colors)

**Comment by simurai** [comment]
Not saying that comments shouldn't be used for documenting, just that it's not the only use case. It actually would be possible to differentiate between documentation comments and line comments:

```less
atom-text-editor::shadow {
  .comment.documentation,
  .comment.documentation .punctuation {
    color: #eeaa75;
  }
}
```

![screen shot 2017-01-18 at 3 45 44 pm](https://cloud.githubusercontent.com/assets/378023/22053628/54aaf00a-dd95-11e6-8698-0051a0945c86.png)

**Comment by simurai** [comment]
Ok, if we don't wanna wait for the next major version of Atom, we could do this:

1. Create a new theme that:
  - Is simple (no settings)
  - Less opinionated and more averaged out
  - Passes the accessibility guidelines (contrast)
  - Keeps performance in mind
  - Is better suited as a template to create your own theme

2. Publish and test it for a while
3. Use it as default theme, but still keep the One themes bundled

Then new users and users that never changed to another theme will see the new one. The rest would have to switch with the theme picker. So negative surprises might still be tolerable.

**Comment by KonradHoeffner** [comment]
The problem is in my opinion that the black in the background isn't true black but grey or a slightly tinted gray, which results in a low contrast and low readability, unlike the default dark themes for terminals such as gnome terminal, which use "true white" (#ffffff) on "true black" (#000000). Even if that is not available as a default theme, it would still be nice to have a theme with such colors as an alternative that is installed by default.

**Comment by cengizhan** [comment]
i also don't like color theme of atom (esp atom dark and one dark). it is really hard to distinguish tabbar from editor window and active tab from other tabs. it is not clear which tab is active from a certain distance.

**Comment by JonasT** [comment]
@simurai unless your codebase is trivial, the line comments being readable can still be quite vital. The only argument I can think of for making line comments barely visible as they are right now is if they are heavily used to temporarily disable code - that however is IMHO quite bad style and not what the display should be optimized for.
